addons/test_mail/tests/test_mail_composer.py

```diff
@@ -1730,109 +1843,149 @@ def test_mail_composer_wtpl_complete(self):
-                                  default_template_id=self.template.id)
-        ))
-        composer = composer_form.save()
-        # ensure some parameters used afterwards
-        author = self.env.user.partner_id
-        self.assertEqual(composer.author_id, author,
-                         'Author cannot be synchronized with a raw email_from')
-        self.assertEqual(composer.email_from, self.template.email_from)
-
-        with self.mock_mail_gateway(mail_unlink_sent=False), \
-             freeze_time(self.reference_now):
-            composer._action_send_mail()
+        for use_domain, scheduled_date, email_layout_xmlid, use_lang in product(
+            (False, True),
+            (False, '{{ (object.create_date or datetime.datetime(2022, 12, 26, 18, 0, 0)) + datetime.timedelta(days=2) }}'),
+            (False, 'mail.test_layout'),
+            (False, True),
+        ):
+            with self.subTest(use_domain=use_domain,
+                              scheduled_date=scheduled_date,
+                              email_layout_xmlid=email_layout_xmlid,
+                              use_lang=use_lang):
+                # update test configuration
+                self.template.write({
+                    'scheduled_date': scheduled_date,
+                })
+                if use_lang:
+                    langs = ('es_ES', 'en_US')
+                    self.test_partners[0].lang = langs[0]
+                    self.test_partners[1].lang = langs[1]
+                else:
+                    langs = (False, False)
+                    self.test_partners.lang = False
 
-            new_partners = self.env['res.partner'].search([
-                ('email', 'in', [email_to_1, email_to_2, email_to_3, email_cc_1])
-            ])
-            self.assertEqual(len(new_partners), 3)
+                ctx = {
+                    'default_model': self.test_records._name,
+                    'default_composition_mode': 'mass_mail',
+                    'default_template_id': self.template.id,
+                }
+                if use_domain:
+                    ctx['default_res_domain'] = [('id', 'in', self.test_records.ids)]
+                else:
+                    ctx['default_res_ids'] = self.test_records.ids
+                if email_layout_xmlid:
+                    ctx['default_email_layout_xmlid'] = email_layout_xmlid
 
-            # global outgoing: emails not sent due to scheduled_date
-            self.assertEqual(len(self._new_mails), 2, 'Should have created 1 mail.mail per record')
-            self.assertEqual(len(self._mails), 0, 'Should not send emails, scheduled in the future')
-            self.assertEqual(self._new_mails.mapped('scheduled_date'),
-                             [self.reference_now + timedelta(days=2)] * 2)
+                # launch composer in mass mode
+                composer_form = Form(self.env['mail.compose.message'].with_context(ctx))
+                composer = composer_form.save()
```

Comment:
Don't really see the point of using all these defaults and a "Form", can't we just create a record?

Or do we need to trigger some onchange or something to make it work?